grakn-graph/src/main/java/ai/grakn/graph/internal/ConceptLog.java

```diff
@@ -40,9 +43,20 @@
  *
  */
 public class ConceptLog {
-    private Set<ConceptImpl> modifiedConcepts;
+
+    //Caches any concept which has been touched before
+    private final Map<ConceptId, ConceptImpl> conceptCache = new HashMap<>();
```

Comment:
Indeed. Low priority improvement though. These caches are transaction specific and will never get too large.